pageserver/src/layered_repository.rs

```diff
@@ -741,8 +743,10 @@ impl Timeline for LayeredTimeline {
         }
 
         let seg = SegmentTag::from_blknum(rel, blknum);
-        let layer = self.get_layer_for_write(seg, rec.lsn)?;
-        self.increase_current_logical_size(layer.put_wal_record(blknum, rec)? * BLCKSZ as u32);
+        let delta_size = self.perform_write_op(seg, rec.lsn, |layer| {
+            layer.put_wal_record(blknum, rec.clone())
```

Comment:
I'd like to avoid these clones if possible

Reply:
I think you could change the function to take a reference to WALRecord, i.e.
```
fn put_wal_record(&self, rel: RelishTag, blknum: u32, rec: &WALRecord) -> Result<()> {
```
And the same to `InMemoryLayer::put_wal_record`. Will need some changes in the callers, but it seems more sensible to pass the WALRecord by reference anyway.